server/api_link.go

```diff
@@ -441,17 +441,29 @@ func (s *ApiServer) LinkGoogle(ctx context.Context, in *api.AccountGoogle) (*emp
 		s.logger.Info("Could not authenticate Google profile.", zap.Error(err))
 		return nil, status.Error(codes.Unauthenticated, "Could not authenticate Google profile.")
 	}
+	
+	displayName := googleProfile.Name
+	if len(displayName) > 255 {
+		// Trim the name in case it is longer than db can store
+		displayName = displayName[0:255]		
+	}
+	
+	avatarUrl := googleProfile.Picture
+	if len(avatarUrl) > 512 {
+		// Ignore the url in case it is longer than db can store
+		avatarUrl = ""
```

Comment:
We should also leave a WARN log to tell developers that this has happened.

Reply:
Agreed, will add.